.travis.yml

```diff
@@ -1,84 +1,69 @@
-
-matrix:
-  include:
-    # gcc on arm64
-    - compiler: gcc
-      arch: arm64
-      addons:
-        apt:
-          packages:
-            - ccache
-      env: &Debug
-        - TRAVIS_CUBERITE_BUILD_TYPE=DEBUG
-    # AppleClang
-    # OSX workers are slower to start up.  Having these first in the build matrix makes travis faster overall.
-    - os: osx
-      compiler: clang
-      before_install:
-        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
-      env: &Release
-        - TRAVIS_CUBERITE_BUILD_TYPE=RELEASE
-    - os: osx
-      compiler: clang
-      env: *Debug
-      before_install:
-        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
-    # Default clang
-    - compiler: clang
-      env: *Release
-    - compiler: clang
-      env: *Debug
-    # clang 3.5
-    - compiler: clang
-      dist: trusty
-      addons: &clang35
-        apt:
-          packages:
-            - clang++-3.5
-            - clang-3.5
-      before_install:
-        - CC=clang-3.5;CXX=clang++-3.5
-      env: *Release
-    - compiler: clang
-      dist: trusty
-      addons: *clang35
-      before_install:
-        - CC=clang-3.5;CXX=clang++-3.5
-      env: *Debug
-    # gcc 4.8
-    - compiler: gcc
-      addons: &gcc48
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-4.8
-            - gcc-4.8
-      before_install:
-        - CC=gcc-4.8;CXX=g++-4.8
-      env: *Release
-    - compiler: gcc
-      addons: *gcc48
-      before_install:
-        - CC=gcc-4.8;CXX=g++-4.8
-      env: *Debug
-
-before_script:
-  - export PATH=$(echo "$PATH" | sed -e 's/:\/usr\/lib\/ccache//')
-script: ./travisbuild.sh
-cache: ccache
-
-
-notifications:
-  email:
-    on_success: change
-    on_failure: always
-branches:
-  only:
-    - coverity_scan
-    - master
+language: cpp
+cache: ccache
+
+# Use Linux by default
+os: linux
+dist: bionic
+
+# CMake version out of date: update
+# TODO: add ARM64, PPC, SPARC builds when we find CMake for them
+addons:
+  snaps:
+  - name: cmake
+    confinement: classic
+    channel: latest
+
+matrix:
+  include:
+    # AppleClang Release
+    # OSX workers are slower to start up.  Having these first in the build matrix makes travis faster overall.
+    - os: osx
+      osx_image: xcode11.3
+      before_install:
+        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
+      env: &Release
+        - TRAVIS_CUBERITE_BUILD_TYPE=RELEASE
+
+    # AppleClang Debug
+    - os: osx
+      osx_image: xcode11.3
+      before_install:
+        - HOMEBREW_NO_AUTO_UPDATE=1 brew install ccache
+      env: &Debug
+        - TRAVIS_CUBERITE_BUILD_TYPE=DEBUG
+
+    # Default Clang Release
+    - compiler: clang
+      cache: false # Fails on precompiled headers otherwise
+      before_install: &use-cmake
+        - export PATH=/snap/bin/:${PATH}
+      env: *Release
+
+    # Default Clang Debug
+    - compiler: clang
+      cache: false # Fails on precompiled headers otherwise

```

Comment:
No it doesn't. We use clang with ccache right now.

Reply:
Oh, the [*pain*](https://travis-ci.org/github/cuberite/cuberite/jobs/682415107#L1004)

![PCH failing on Clang](https://user-images.githubusercontent.com/2939130/81091323-f9aa2d80-8ef6-11ea-9ec0-5b2c1d3fef99.png)

All the C++17 PRs need to be merged together, so PCH is going to be enabled on all platforms.

Reply:
Try it again with `travisbuild.sh`.  There were some clang-specific issues that required `CCACHE_CPP2` to be defined.

Reply:
Also, ccache is a much bigger win than pch so I would disable pch instead of ccache.

Reply:
OK. I can add a Travis-specific flag if the shell script doesn't work, but in order to test I'll need everything merged, the see-makes are in another PR xD